backend/oauth.go

```diff
@@ -0,0 +1,236 @@
+// Copyright Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backend
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"net/http"
+
+	"golang.org/x/net/context"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/appengine"
+)
+
+const (
+	OAUTH_BASE = "/oauth/"
+
+	OAUTH_COOKIE = "oauth2_cookie"
+
+	CLIENT_SECRET_GOOGLE_FILENAME = "google_client_secret.json"
+)
+
+var (
+	oauthScopes       = []string{"openid", "profile"}
+	oauthGoogleConfig *oauth2.Config
+)
+
+func InitOAuth() {
+	RegisterHandler(OAUTH_BASE+"google/login", oauthGoogleLogin)
```

Comment:
nit: I'd change the segments to `login/google` (same with callback and method names). This way your path goes from generic -> specific.